src/registrar/forms/domain.py

```diff
@@ -529,8 +529,7 @@ def save(self, commit=True):
             elif self.is_tribal and not self._field_unchanged("organization_name"):
                 raise ValueError("organization_name cannot be modified when the generic_org_type is tribal")
 
-        else:
-            super().save()
+        super().save()
```

Comment:
When self.has_changed was evaluating to True, super.save was never called, which never allowed the audit log to be created.  This fixes the first bug